tools/applicationinsights-web-snippet/src/snippet.ts

```diff
@@ -235,7 +247,18 @@ declare var cfg:ISnippetConfig;
                 return scriptElement;
             }
 
-            let theScript = _createScript();
+            let theScript = _createScript(targetSrc);
+
+            if (appInsights.cr && loadFailed && domains.some(endpoint => targetSrc.indexOf(endpoint) !== -1)){
+                for (const currentEndpoint of domains) {
```

Comment:
We should also use for (lp = 0 ; lp < ...) rather than "of" as some pages frameworks have extensions installed which causes this to not do what we expect (like iterate into other "manually" added functions via the Array.prototype -- again causing this to throw)